app/models/user.rb

```diff
@@ -7,6 +9,9 @@ class User < ActiveRecord::Base
   validates :password, confirmation: true, on: :create
   has_secure_password(validations: false)
 
+  before_create { generate_token(:uid) if Proc.new { |u| u.provider == 'email'} }
```

Comment:
why are you doing a proc here? The one below doesn't have a proc and it works...